libsnowflakeclient/deps-build/darwin/azure/include/http/libcurl_http_client.h

```diff
@@ -327,8 +327,11 @@ namespace azure {  namespace storage_lite {
                 {
                   curl_easy_setopt(h, CURLOPT_PROXY, proxy_host.c_str());
                   curl_easy_setopt(h, CURLOPT_PROXYPORT, (long) proxy_port);
-                  curl_easy_setopt(h, CURLOPT_PROXYUSERNAME, proxy_user.c_str());
-                  curl_easy_setopt(h, CURLOPT_PROXYPASSWORD, proxy_password.c_str());
+                  if (!proxy_user.empty() || !proxy_password.empty())
+                  {
```

Comment:
we will end up using empty value if one and only one of proxy_user and proxy_password is empty, is it the intended behavior? if that's indeed the case, please add comment to document it. If not, it's introducing a bug.

Reply:
@sfc-gh-jtang 
Actually this is discussed in the original PR in libsnowflakeclient
https://github.com/snowflakedb/libsnowflakeclient/pull/504
The point is:
- It's the best effort to represent the customer settings for authentication. For an example if the customer proxy can take username with empty password (I don't think any customer will do that but who knows), if we don't send out Proxy-Authorization Header with the user name and empty password, that would be a driver bug. But in the case customer proxy can't take that, we can say it's customer setting issue as it's the customer to set proxy username but not set password.
- It's copying the behavior from aws sdk for fixing similar issue: https://github.com/aws/aws-sdk-cpp/commit/59fe5b1b4d3cf9d26586a543376818e581f5aafc

Reply:
@sfc-gh-jtang 
This is the intended behavior. But to document it in the source comment, it should be made on libsnowflakeclient side and I can do that later. Here just the copy for all artifacts and header files from libsnowflakeclient.

Reply:
are you saying the only case the logic is guarding against is both username and password are empty? if that's the case, it's better write the code as if (! (proxy_user.empty() && proxy_password.empty()). If that's indeed the case, please make the change with the comment, I will approve the change afterwards. Thanks

Reply:
and if the default is empty, then we don't really need to check for empty value, is it right?

Reply:
@sfc-gh-jtang 
after all, if any change needs to be made, it should be made in libsnowflakeclient https://github.com/snowflakedb/libsnowflakeclient/blob/v0.6.20/deps/azure-storage-cpplite-0.1.20/include/http/libcurl_http_client.h#L330
Here is a copy from the latest libsnowflakeclient and changing the header file here won't really change the behavior.

I'll make a PR on libsnowflakeclient side to change the code to 
if (! (proxy_user.empty() && proxy_password.empty())
and add comment there as well. But since it doesn't change the behavior it should not be a release blocker.

The default is empty, but it's curl's behavior that it only send out Proxy-Authorization header when either username or password is set. So yes we just need to guard against that in case both username and password are empty.

Thanks,
Harry

Reply:
So the acceptable behavior is either 1) none of the user/password curl options are set 2) both are set when at least one of them are not empty. This behavior is not straightforward. Please document this behavior clearly in the libsnowflakeclient and update this PhP header file during the next sync. I will approve this PR for now.